### ISSUE TITLE
feat: Query tagging for targets

### ISSUE THREAD

**Issue by qbatten** [meltano/sdk#1804]
### Feature scope

Targets (data type handling, batching, SQL object generation, tests, etc.)

### Description

It'd be super useful for me if we could add query tags for targets; specifically I want to be able to add query tags to my snowflake target (I'm using transferwise/target-snowflake and it supports this).

Transferwise's snowflake target has this implemented. Could be a good place to start. They set it on the session params [here](https://github.com/transferwise/pipelinewise-target-snowflake/blob/aa6ac2e392b6148994f9138241630ba8d684adac/target_snowflake/db_sync.py#L302C13-L302C31).

Very loosely speccing out what this could look like...
- [SQLAlchemy](https://docs.sqlalchemy.org/en/20/core/engines.html#sqlalchemy.create_engine.params.connect_args) lets you add arbitrary connect_args and it looks like [sqlalchemy-snowflake](https://github.com/snowflakedb/snowflake-sqlalchemy) (cmd-f "connect_args" on that page) does support arbitrary Snowflake session params.
- The SQLConnector would have to be the object handling this. It could check the config for e.g. a "connect_args" argument that it then passes [to the sqlalchemy engine](https://github.com/meltano/sdk/blob/9c0c9602974042511105d2bc80b29141feb79543/singer_sdk/connectors/sql.py#L319).
- So, in summary, SQLConnector could check for connect_args in config, then pass those args to it's create_engine statement. For Snowflake this would look like the target having a nested config option: connect_args > query_tag.

**Comment by tayloramurphy** [comment]
@qbatten great idea. I'd definitely support this if you or someone is willing to submit a PR. Do you know if other warehouses / DBs support something like htis?

**Comment by edgarrmondragon** [comment]
Created https://github.com/MeltanoLabs/target-snowflake/issues/153 before rediscovering this issue.

`QUERY_TAG` seems specific to snowflake, but we could change the scope of this issue to make it easier for SQL taps and targets to pass engine-specific parameters to the underlying connection.
